packages/core/admin/ee/server/services/audit-logs.js

```diff
@@ -99,10 +117,40 @@ const createAuditLogsService = (strapi) => {
 
   return {
     async register() {
-      const retentionDays =
-        features.get('audit-logs')?.options.retentionDays ?? DEFAULT_RETENTION_DAYS;
+      // Handle license being enabled
```

Comment:
When not using classes, you should avoid using the `this` and use an internal state in your service 

```js
const createAuditLogsService = (strapi) => {

const state = {  };


return {
  register() {
    if (!state.eeEnableUnsubscribe) {
      state.eeEnableUnsubscribe  = ...
    }
  }

} 

}
```

This also has the advantage of really enforcing private vars as you never return the internal state of the service